File: sdk/python-sdk/verity_sdk/protocols/v1_0/Connecting.py
from verity_sdk.protocols.Protocol import Protocol
from verity_sdk.utils import COMMUNITY_MSG_QUALIFIER


class Connecting(Protocol):
    """
    The Connecting protocols form secure, private and self-sovereign channel between two independent parties.
    This protocol facilitates the exchange of keys and endpoints that will be used in all future interactions.
    This connecting process starts with the Relationship protocol which provisions a relationship on the Verity
    Application and creates the an invite. With the invite, this Connecting protocol can be started and completed.
    """

    MSG_FAMILY = 'connections'
    """the family name for the message family"""
    MSG_FAMILY_VERSION = '1.0'
    """the version for the message family"""

    REQUEST_RECEIVED = 'request-received'
    """Name for 'request-received' signal message"""
    RESPONSE_SENT = 'response-sent'
    """Name for 'response-sent' signal message"""
    ACCEPT_INVITE = 'accept'
    """Name for 'accept' control message"""
    STATUS = 'status'
    """Name for 'status' control message"""

    def __init__(self,
                 for_relationship: str,
                 thread_id: str = None,
                 label: str = None,
                 base64_invite_url: str = None):
        """
        Args:
            for_relationship (str): The relationship identifier (DID) to use for the connections exchange. Normally,
                its relationship will have been created as a reaction to receiving the invitation.
            thread_id (str): The thread id of the already started protocol
            label (str): A human readable string that will label the caller identity (often an organization).
                E.g. 'Acme Corp`
            base64_invite_url (str): the invitation URL as specified by the Aries 0160: Connection Protocol
                (eg. https://<domain>/<path>?c_i=<invitation-string>)
        """
        super().__init__(
            self.MSG_FAMILY,
            self.MSG_FAMILY_VERSION,
            msg_qualifier=COMMUNITY_MSG_QUALIFIER,
            thread_id=thread_id
        )

        self.for_relationship = for_relationship
        self.label = label
        self.base64_invite_url = base64_invite_url

    async def accept(self, context):
        """
        Accepts connection defined by the given invitation

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent
        """
        await self.send_message(context, await self.accept_msg_packed(context))

    def accept_msg(self, context):
        """
        Creates the control message without packaging and sending it.

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent

        Return:
            the constructed message (dict object)
        """
        msg = self._get_base_message(self.ACCEPT_INVITE)
        self._add_thread(msg)
        self._add_relationship(msg, self.for_relationship)
        msg['label'] = self.label
        msg['invite_url'] = self.base64_invite_url
        return msg

    async def accept_msg_packed(self, context):
        """
        Creates and packages message without sending it.

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent

        Return:
            the bytes ready for transport
        """
        return await self.get_message_bytes(context, self.accept_msg(context))

    async def status(self, context):
        """
        Sends the get status message to the connection

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent
        """
        await self.send_message(context, await self.status_msg_packed(context))

    def status_msg(self, context):
        """
        Creates the control message without packaging and sending it.

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent

        Return:
            the constructed message (dict object)
        """
        msg = self._get_base_message(self.STATUS)
        self._add_thread(msg)
        self._add_relationship(msg, self.for_relationship)
        return msg

    async def status_msg_packed(self, context):
        """
        Creates and packages message without sending it.

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent

        Return:
            the bytes ready for transport
        """
        return await self.get_message_bytes(context, self.status_msg(context))

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/Conecting/Connecting.cs
using System;

namespace VeritySDK
{
    
/// <summary>
    /// Factory for Connecting protocol objects.
    /// <p/>
    /// The Connecting protocols form secure, private and self-sovereign channel between two independent parties.
    /// This protocol facilitates the exchange of keys and endpoints that will be used in all future interactions.
    /// This connecting process starts with the Relationship protocol which provisions a relationship on the Verity
    /// Application and creates the an invite. With the invite, this Connecting protocol can be started and completed.
    /// </summary>
    /// <see cref="Relationship"/>
    /// <see cref="https://github.com/hyperledger/aries-rfcs/tree/9b0aaa39df7e8bd434126c4b33c097aae78d65bf/features/0160-connection-protocol"/>
    public class Connecting
    {
        /// <summary>
        /// Simple constructor
        /// </summary>
        private Connecting() { }

        
/// <summary>
        /// Constructor for the 1.0 Connections object. This constructor creates an object that is ready to accept an invitation and start the Connections protocol.
        /// </summary>
        /// <param name="forRelationship">The relationship identifier (DID) to use for the connections exchange. Normally, its relationship will have been created as a reaction to receiving the invitation.</param>
        /// <param name="label">A human readable string that will label the caller identity (often an organization). E.g. 'Acme Corp`</param>
        /// <param name="base64InviteURL">the invitation URL as specified by the Aries 0160: Connection Protocol (eg. https://<domain>/<path>?c_i=<invitation-string>)</param>
        /// <returns>1.0 Connections object</returns>
        public static ConnectionsV1_0 v1_0(String forRelationship, String label, String base64InviteURL)
        {
            return new ConnectionsImplV1_0(forRelationship, label, base64InviteURL);
        }

        /// <summary>
        /// Constructor for the 1.0 Connections object. This constructor re-creates an object from a known relationship and threadId. This object can only check status of the protocol.
        /// </summary>
        /// <param name="forRelationship">the relationship identifier (DID) to use for the connections protocol.</param>
        /// <param name="threadId">the thread id of the already started protocol</param>
        /// <returns>1.0 Connections object</returns>
        public static ConnectionsV1_0 v1_0(String forRelationship, String threadId)
        {
            return new ConnectionsImplV1_0(forRelationship, threadId);
        }
    }
}
